utils/challenges/challenges.js

```diff
@@ -0,0 +1,36 @@
+import firebase from "../firebase/firebase";
+
+export async function setCurrentChallenge(teamId) {
+  // pick a random challenge
+  const challengeCountSnapshot = await firebase
+    .database()
+    .ref("/challenges/count")
+    .once("value");
+  const challengeCount = challengeCountSnapshot.val();
+  // challenge ids are indexed at 0
+  const challengeId = Math.floor(Math.random() * challengeCount);
+
+  const assignedTime = firebase.database.ServerValue.TIMESTAMP;
+  // time since the Unix epoch, in milliseconds) as determined by the Firebase servers
+  // see https://firebase.google.com/docs/reference/js/firebase.database.ServerValue
+
+  const newChallenge = {
+    challengeId,
+    teamId,
+    assignedTime
+  };
+
+  const assignedChallengeRef = await firebase
+    .database()
+    .ref(`/assignedChallenges/`)
```

Comment:
Did we decide to make assignedChallenges a separate object?

Reply:
The objects will not be static since we need to update submissions to the assigned challenge. I'm still open to changing it, but I didn't want to make archived challenges too complicated (a list of assigned challenge id's instead of the full challenge). There's pros and cons to both that we can discuss on Tuesday or just change as the need arises.